### PR TITLE
Implementing build for intel based macOS

### PR DESCRIPTION
**Pull request by 3c7**
As a side effect of #67 and affecting #14 partly, this pull request includes a few changes allowing to build yari on macOS x64. An example run can be seen here: https://github.com/3c7/yari/actions/runs/7396353584.

Somehow bindgen is able to derive `Debug` for `YR_SCAN_CONTEXT` itself already, so the implementation in yari-sys/src/bindings.rs was limited to linux and windows systems. Additionally, `link_linux()` was renamed to `link_unix()` and limited to both, linux and macos.

In order to be able to use the bundled bindings, another set of bindings for macos was provided.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -43,3 +41,26 @@ jobs:
         pip install maturin pytest
         maturin develop
         pytest tests
+build-macos:
```
Comment (MatejKastak) [inline .github/workflows/test.yml:44]: I think that you are missing indentation, which is the reason why the action fails currently: https://github.com/avast/yari/actions/runs/7396538047
```suggestion
  build-macos:
```

**Review by MatejKastak — CHANGES_REQUESTED** [review]
Thank you very much for the changes. There are two small things to fix after that we can merge.

File: .github/workflows/test.yml (hunk 1 above)
Comment (3c7) [reply .github/workflows/test.yml:44]: Oops!

**Comment by 3c7** [comment]
Implemented the changes and the workflow runs now. :)

**Review by MatejKastak — APPROVED** [review]
Thank you very much. LGTM :+1:

**Pull request merged**